pemFioi/quickAlgo/blockly_blocks.js

```diff
@@ -2525,6 +2525,32 @@ function getBlocklyBlockFunctions(maxBlocks, nbTestCases) {
          return true;
       },
 
+      /**
+       * This method allow us to check if there are any non-existent block that are used inside of the xml given
+       * @param xml The code of the user
+       * @return An {Array} containing all ids of blocks that are not allowed but used.
+       */
+      checkNonexistentBlocks: function(xml) {
```

Comment:
Instead of copy-pasting the checkBlocksAreAllowed function, it would be better to modify checkBlocksAreAllowed into a more generic function which fulfills both purposes, and then have multiple possible result behaviors.

Reply:
I have merged both functions in next commit